app/src/main/java/com/medmapper/v33001/dao/MedicineDAO.kt

```diff
@@ -6,12 +6,17 @@ import kotlinx.coroutines.flow.Flow
 
 @Dao
 interface MedicineDAO {
+
     @Query("SELECT * FROM Medicine ORDER BY name ASC")
     fun getAlphabetizedMedicine(): Flow<List<Medicine>>
 
     @Insert(onConflict = OnConflictStrategy.IGNORE)
     suspend fun insert(medicine: Medicine)
 
-    @Query("DELETE FROM Medicine")
-    suspend fun deleteAll()
+    @Delete
+    suspend fun delete(medicine: Medicine)
+
+    @Update
+    suspend fun update(medicine: Medicine)
+
```

Comment:
Room provides specific annotations for DELETE and UPDATE. Using @Delete and @Update annotations instead of @Query can make the code more concise and easier to read.

Source: https://developer.android.com/training/data-storage/room/accessing-data